app/src/main/java/com/example/myphonebook/PhoneBookEntry.java

```diff
@@ -0,0 +1,56 @@
+package com.example.myphonebook;
+
+import android.net.Uri;
+import android.util.Log;
+
+public class PhoneBookEntry implements Entry {
+    private String Name,Number;
+
+    EMailProvider eMailProvider;
+    PictureProvider pictureProvider;
+
+    public PhoneBookEntry(String Name, String Number, String email, String picUriStr){
```

Comment:
why not simply hold this 4 arguments as members? why do you need to wrap them with providers?